promql/engine.go

```diff
@@ -116,6 +116,9 @@ type Query interface {
 	Stats() *stats.QueryTimers
 	// Cancel signals that a running query execution should be aborted.
 	Cancel()
+	// EnablePartialResponse enables subsequent Exec invokes to return partial responses without error.
```

Comment:
Why does this require any changes in the engine? The user of the code can already check for warnings.

This is also worded as though enabling is not the typical situation, which is misleading.